### ISSUE TITLE
PIP --use-feature=2020-resolver to resolve package conflicts

### ISSUE THREAD

**Issue by charliekang** [conan-io/conan-docker-tools#230]
### Description of Problem, Request, or Question
PIP change in resolving packaging dependencies
```
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

requests 2.24.0 requires urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1, but you'll have urllib3 1.26.0 which is incompatible.
```
### Environment Details
* Conan Docker Tools image: **conanio/gcc9**
* Operating System: **Linux**
* Operation System Version: **Ubuntu 18.04.5**
* Python version (If you are building some image): **python 3.8.6**

### Steps to reproduce (Include if Applicable)
- clone repo : commit f1df0b92643dce4fe19c8b13a7cfbc28a877ddb3
- cd into dir and build docker image (part 1)

  Image build completes, check version:
- run conan --version (part 2)

### Build logs (Include if Available)
**Part 1**
```
# docker build conan_gcc9:test .

...
update-alternatives: using /opt/pyenv/bin/pyenv to provide /usr/bin/pyenv (pyenv) in auto mode
Downloading Python-3.7.5.tar.xz...
-> https://www.python.org/ftp/python/3.7.5/Python-3.7.5.tar.xz
Installing Python-3.7.5...
Installed Python-3.7.5 to /opt/pyenv/versions/3.7.5

ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

requests 2.24.0 requires urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1, but you'll have urllib3 1.26.0 which is incompatible.
...

```

**Part 2**
```
# docker run --rm -it conan_gcc9:test conan --version

Traceback (most recent call last):
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (urllib3 1.25.5 (/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages), Requirement.parse('urllib3!=1.25.4,!=1.25.5'), {'conan'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/pyenv/versions/3.7.5/bin/conan", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3250, in <module>
    @_call_aside
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3234, in _call_aside
    f(*args, **kwargs)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3263, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'urllib3!=1.25.4,!=1.25.5' distribution was not found and is required by conan
```

**Comment by charliekang** [comment]
Currently can append this line in Dockerfile

```
- && pip install -q --no-cache-dir conan conan-package-tools
+ && pip install -q --no-cache-dir --use-feature=2020-resolver conan conan-package-tools
```

**Comment by uilianries** [comment]
Hi @charliekang The error related to urllib3 has been solved by Conan 1.31.2: https://docs.conan.io/en/latest/changelog.html#nov-2020

A new Docker image should available soon. Regards
